docs/oauth-setup.md

```diff
@@ -0,0 +1,143 @@
+# Using an external OAuth provider with S.H.I.E.L.D
+
+## Github
+
+### Open the Github settings page
+
+![Github Config Image](https://raw.githubusercontent.com/starkandwayne/shield/oauth-docs/docs/images/githubAuth/github1.png "github OAuth")
+
+![Github Config Image](https://raw.githubusercontent.com/starkandwayne/shield/oauth-docs/docs/images/githubAuth/github2.png "github OAuth")
+
+### Open "OAuth Apps" under the "Developer settings" field on the lower left menu
+
+![Github Config Image](https://raw.githubusercontent.com/starkandwayne/shield/oauth-docs/docs/images/githubAuth/github3.png "github OAuth")
+
+### Register a new Github OAuth client with the "Register a new application" button
+
+![Github Config Image](https://raw.githubusercontent.com/starkandwayne/shield/oauth-docs/docs/images/githubAuth/github4.png "github OAuth")
+
+- The only field that is SHIELD specific here is the "Authorization callback URL" field, which needs to be:
+  - `http://<YOUR_SHIELD_ADDRESSS_HERE>/auth/oauth/<YOUR_GH_IDENTIFIER_FROM_YOUR_SHIELD_CONFIG_HERE>/redir`
+- The other fields can contain any information that you'd like, as this client will be specific to your instance of SHIELD
+- Github will now generate a client id and client secret that you will need to copy to the config yaml file you feed to SHIELD.
```

Comment:
Make this less colloquial, something like "... that you will need to put in the SHIELD configuration filee"